### ISSUE TITLE
Should mailalias_core be declared as a dependency ?

### ISSUE THREAD

**Issue by mfaure** [camptocamp/puppet-postfix#236]
Hi,

(I'm quite new to puppet, so excuse me if the following is wrong or incorrectly explained :) )

With Puppet 6, I tried to install Postfix on a Ubuntu 18.04 node (with a bare `include postfix`). On the node when running `puppet agent -t`, I had the following error:

```
Info: Loading facts
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Unknown resource type: 'mailalias' (file: /etc/puppetlabs/code/environments/production/modules/postfix/manifests/files.pp, line: 104, column: 5) on node mynode.taqamaqa.com
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```

I got rid of the problem once I added [mailalias_core](https://forge.puppet.com/puppetlabs/mailalias_core) to my puppetfile.

So from a documentation point of view, should *mailalias_core* be declared as a dependency ?

**Comment by mfaure** [comment]
After some tchat with a colleague of mine, this looks mandatory since Puppet 6. See [Types not repackaged into Puppet agent](https://puppet.com/docs/puppet/6.0/type.html#types-not-repackaged-into-puppet-agent)

**Comment by Zamiell** [comment]
Also just bumped into this error. Thanks for the quick fix mfaure.

**Comment by raphink** [comment]
https://github.com/camptocamp/puppet-postfix/pull/242 solves this

**Comment by anarcat** [comment]
this ticket can be closed.

**Comment by raphink** [comment]
Fixed in #242
